app/src/app_sht40.c

```diff
@@ -67,3 +68,54 @@ int app_sht40_read(float *temperature, float *humidity)
 
 	return 0;
 }
+
+int app_sht40_read_serial(uint32_t *serial_number)
+{
+	int ret;
+
+	const struct device *dev = DEVICE_DT_GET(DT_NODELABEL(sht40));
+	if (!device_is_ready(dev)) {
+		LOG_ERR("Device not ready");
+		return -ENODEV;
+	}
+
+	/* Get I2C bus and address from devicetree */
+	const struct device *i2c_dev = DEVICE_DT_GET(DT_BUS(DT_NODELABEL(sht40)));
+	if (!device_is_ready(i2c_dev)) {
+		LOG_ERR("I2C device not ready");
+		return -ENODEV;
+	}
+
+	uint16_t addr = DT_REG_ADDR(DT_NODELABEL(sht40));
+
+	/* SHT4x (SHT40/SHT43) Read Serial Number command: 0x89
+	 * Note: For SHT30/SHT33 use 2-byte command 0x3780 instead
+	 */
+	uint8_t cmd = 0x89;
+	uint8_t data[6];
+
+	ret = i2c_write(i2c_dev, &cmd, 1, addr);
+	if (ret) {
+		LOG_ERR_CALL_FAILED_INT("i2c_write", ret);
+		return ret;
+	}
+
+	/* Wait for measurement (1ms should be enough for serial read) */
+	k_sleep(K_MSEC(1));
+
+	ret = i2c_read(i2c_dev, data, sizeof(data), addr);
+	if (ret) {
+		LOG_ERR_CALL_FAILED_INT("i2c_read", ret);
+		return ret;
+	}
+
+	/* Serial number is in data[0:1] and data[3:4], with CRC in data[2] and data[5] */
+	if (serial_number) {
+		*serial_number = ((uint32_t)data[0] << 24) | ((uint32_t)data[1] << 16) |
+				 ((uint32_t)data[3] << 8) | (uint32_t)data[4];
+	}
+
+	LOG_DBG("SHT40 Serial: 0x%08X", *serial_number);
```

Comment:
Is it desired to print the output in HEX? How is the serial number presented by Sensirion?